From cf5934043170af8c404b998dc3c5de2aa3276198 Mon Sep 17 00:00:00 2001
From: Steve Loughran <stevel@apache.org>
Date: Tue, 17 May 2016 13:18:51 +0100
Subject: [PATCH 1856/2848] HADOOP-13158 S3AFileSystem#toString might throw
 NullPointerException due to null cannedACL.
 (Chris Nauroth via stevel)

(cherry picked from commit 42d9beaa1db6989858e3e91e8a7e6d00799119b1)

Change-Id: I56b657f9cde75a89ca5cbd64a698e54846689f66
---
 .../org/apache/hadoop/fs/s3a/S3AFileSystem.java    |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java
index 2198d9f..9c1ac8c 100644
--- a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java
+++ b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java
@@ -1497,7 +1497,9 @@ public String toString() {
     sb.append(", partSize=").append(partSize);
     sb.append(", enableMultiObjectsDelete=").append(enableMultiObjectsDelete);
     sb.append(", maxKeys=").append(maxKeys);
-    sb.append(", cannedACL=").append(cannedACL.toString());
+    if (cannedACL != null) {
+      sb.append(", cannedACL=").append(cannedACL.toString());
+    }
     sb.append(", readAhead=").append(readAhead);
     sb.append(", blockSize=").append(getDefaultBlockSize());
     sb.append(", multiPartThreshold=").append(multiPartThreshold);
-- 
1.7.9.5

